aicsimageio/utils/cached_property.py

```diff
@@ -0,0 +1,62 @@
+from typing import Optional, Callable, Any
+
+try:
+    from functools import cached_property
+except ImportError:
+    from threading import RLock
+
+    _NOT_FOUND = object()
+
+    # backport from 3.8
```

Comment:
yep, honestly this is optional too (I could remove it if you want and we can just wait for 3.8).  It's seems like a pretty handy util for this library, since there are a lot of lazy props that only needed to be calculated once.

Reply:
I think we would rather have this earlier and still support 3.7 but agree this needs a louder comment saying why it exists